ui/lib/kmip/addon/routes/role.js

```diff
@@ -1,26 +1,26 @@
 import Route from '@ember/routing/route';
 import { inject as service } from '@ember/service';
 
-export default Route.extend({
-  store: service(),
-  secretMountPath: service(),
-  pathHelp: service(),
+export default class KmipRoleRoute extends Route {
+  @service store;
+  @service secretMountPath;
+  @service pathHelp;
 
-  beforeModel() {
-    return this.pathHelp.getNewModel('kmip/role', this.secretMountPath.currentPath);
-  },
+  async beforeModel() {
+    return await this.pathHelp.getNewModel('kmip/role', this.secretMountPath.currentPath);
+  }
 
-  model(params) {
-    return this.store.queryRecord('kmip/role', {
+  async model(params) {
+    return await this.store.queryRecord('kmip/role', {
       backend: this.secretMountPath.currentPath,
       scope: params.scope_name,
       id: params.role_name,
     });
-  },
+  }
 
-  setupController(controller) {
-    this._super(...arguments);
```

Comment:
We should probably still call super, but in glimmer world I think it's `super.setupController(...arguments)`

Reply:
I got an error when I tried with the super.setupController(...arguments) and then looked at the docs and they don't mention the super. I'll try again.